editor/hooks/copy-content/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { ClipboardButton, withState } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+import { query } from '@wordpress/data';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+
+function CopyContentButton( { editedPostContent, hasCopied, setState } ) {
+	return (
+		<ClipboardButton
+			text={ editedPostContent }
+			className="components-menu-items__button"
+			onCopy={ () => setState( { hasCopied: true } ) }
+			onFinishCopy={ () => setState( { hasCopied: false } ) }
+		>
+			{ hasCopied ?
+				__( 'Copied!' ) :
+				__( 'Copy All Content' ) }
+		</ClipboardButton>
+	);
+}
+
+const Enhanced = compose(
+	query( ( select ) => ( {
+		editedPostContent: select( 'core/editor', 'getEditedPostContent' ),
+	} ) ),
+	withState( { hasCopied: false } )
+)( CopyContentButton );
```

Comment:
Trying to look at this file as a prospective extension author, considering readability, etc. I toyed with

```js
const applyGlobalState = query( ( select ) => ( {
  …
} ) );

const applyLocalState = withState( { hasCopied: false } );

const Enhanced = compose(
  applyGlobalState,
  applyLocalState
)( CopyContentButton );
```

but I wasn't sure it improved things. At some point, more labels (such as function identifiers) may become a hinderance, and this felt slightly awkward.

Reply:
It's always a personal preference and really depends on the particular use case. It is fine as it is because those HOC applied are short. I usually extract variables when you need to scroll the screen to discover that a given code is part of the `compose` or `connect`.